src/components/ItemModalBackground.tsx

```diff
@@ -0,0 +1,201 @@
+    if (index !== -1) {
+      setSelectedTags(selectedTags.filter((_, i) => i !== index));
+    } else {
+      setSelectedTags([...selectedTags, tag]);
+    }
+  };
+  useEffect(() => {
+    setFormData({
+      ...formData,
+      Categories: selectedTags,
+    });
+  }, [selectedTags]);
+
+  return (
+    <div className="modalBackground">
+      <div className="modalContainer">
+        <div className="titleCloseBtn">
+          <button type="button" onClick={setOpenModal}>
+            X
+          </button>
+        </div>
+        <div className="title">
+          <h1>Add Item</h1>
+        </div>
+        <div>
+          {/* Textfields */}
+          <Box>
+            <div style={{ display: "flex", justifyContent: "center" }}>
+              <TextField
+                name="Name"
+                className="textField"
+                sx={{
+                  width: 550,
+                  paddingBottom: 2,
+                }}
+                label="Name"
+                onChange={handleInputChange}
+              />
+            </div>
+            <div
+              style={{
+                display: "flex",
+                justifyContent: "center",
+              }}
+            >
+              <div
+                style={{
+                  display: "flex",
+                  paddingRight: 30,
+                }}
+              >
+                <TextField
+                  name="Price"
+                  className="textField"
+                  sx={{
+                    width: 260,
+                  }}
+                  label="Price"
+                  onChange={handleInputChange}
+                />
+              </div>
+
+              {/* Date picker */}
+              <LocalizationProvider dateAdapter={AdapterDayjs}>
+                <DatePicker
+                  sx={{
+                    width: 260,
+                  }}
+                  disableFuture
+                  label="Date Added"
+                  onChange={(newValue) => {
+                    setFormData({ ...formData, DateAdded: newValue });
+                  }}
+                />
+              </LocalizationProvider>
+            </div>
+          </Box>
+
+          {/* Selected Buttons indicator */}
+          <div>
+            <div
+              style={{
+                display: "flex",
+                flexDirection: "row",
+                alignItems: "center",
+                paddingBottom: 20,
+                paddingTop: 20,
+              }}
+            >
+              <div style={{ marginRight: 10 }}>
+                <h3>Select Tags</h3>
+              </div>
+              <div>
+                <button
+                  type="button"
+                  className={selectedTags.length > 0 ? "tag" : "tagSelected"}
+                >
+                  {selectedTags.length} selected
+                </button>
+              </div>
+            </div>
+
+            {/* Category Selectable Buttons */}
+            <div
```

Comment:
Refactor this to make use of javascript's map. We make use of it in TagsList.tsx